app/controllers/users/logix_controller.rb

```diff
@@ -1,28 +1,49 @@
 class Users::LogixController < ApplicationController
-  before_action :authenticate_user! ,only: [:edit, :update, :groups]
+  before_action :authenticate_user!, only: [:edit, :update, :groups]
+  before_action :set_user, except: [:typeahead_educational_institute]
+
   def index
-    @user = User.find(params[:id])
     @edit_access = (user_signed_in? and current_user.id == @user.id)
   end
+
   def favourites
-    @user = User.find(params[:id])
     @projects = @user.rated_projects
   end
+
   def profile
-    @profile = User.find(params[:id])
+    @profile = @user
   end
+
   def edit
-    @profile = current_user
+
   end
+
+  def typeahead_educational_institute
+    query = params[:query]
+    educational_institute_list = User.where("educational_institute LIKE ?", "%#{query}%").uniq { |user| user.educational_institute }
+                                     .pluck(:educational_institute)
```

Comment:
This query can be expensive, uniq is an array operation, not a db operation. do uniqueness check in database query itself

Reply:
@tachyons, so i have fixed this by replacing uniq with distinct in my latest commit.